Python/ceval.c

```diff
@@ -5372,23 +5380,23 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             if (res == NULL) {
                 goto error;
             }
+            next_instr += _PyOpcode_InlineCacheEntries[BINARY_OP];
             DISPATCH();
         }
 
         TARGET(BINARY_OP_ADAPTIVE) {
             assert(cframe.use_tracing == 0);
-            SpecializedCacheEntry *cache = GET_CACHE();
-            if (cache->adaptive.counter == 0) {
+            uint16_t *counter = inline_cache_uint16(next_instr, 0);
```

Comment:
Given that cache entries must be both the same size and alignment as instructions, it makes sense to use `_Py_CODEUNIT` here.